client/src/index.ts

```diff
@@ -384,16 +384,9 @@ export class RestRepository<T> extends Repository<T> {
     super(null, table, {});
     this.client = client;
 
-    const doWeHaveFetch = typeof fetch !== 'undefined';
-    const isInjectedFetchProblematic = !this.client.options.fetch;
-
-    if (doWeHaveFetch) {
-      this.fetch = fetch;
-    } else if (isInjectedFetchProblematic) {
-      throw new Error(errors.falsyFetchImplementation);
-    } else {
-      this.fetch = this.client.options.fetch;
-    }
+    const fetchImpl = typeof fetch !== 'undefined' ? fetch : this.client.options.fetch;
+    if (!fetchImpl) throw new Error('No fetch implementation provided');
```

Comment:
I see some value in this check:

```ts
    if (!fetchImpl && 'fetch' in this.client.options) throw new Error('Invalid fetch implementation provided');
```

That was the idea with the "falsy" error message.

In any case we should remove the `errors.falsyFetchImplementation` in the errors file if we don't use it anymore.

Reply:
Since it's not related to the fix on CF, let's move the discussion to #56